onvm/onvm_mgr/onvm_init.c

```diff
@@ -392,6 +399,11 @@ init_port(uint8_t port_num) {
 
         rte_eth_promiscuous_enable(port_num);
 
+        if (ONVM_USE_JUMBO_FRAMES)
+                rte_eth_dev_set_mtu(port_num, 9000);
```

Comment:
Is this configuration enough for enabling jumbo packet support? I see some rx port configuration `DEV_RX_OFFLOAD_JUMBO_FRAME` in this Feature list [1]? Also, you can check the usage of this configuration in [2] example.

[1] https://dpdk.readthedocs.io/en/v17.11/nics/features.html
[2] https://github.com/DPDK/dpdk/blob/main/examples/ipsec-secgw/ipsec-secgw.c#L2175

Reply:
This flag offloads some handling of the jumbo frame to the physical NIC. I'm guessing what it does is forces the hardware to assemble the jumbo frame instead of the PMD (correct me if I'm wrong @twood02 ). This could be used to increase performance. I don't think it's necessary to enable this flag in order to support receiving jumbo frames. We should check if the NIC PMD we are using actually supports the MTU that is passed in. DPDK will tell you if the PMD doesn't have a set_mtu function or if it doesn't support the MTU: https://github.com/DPDK/dpdk/blob/main/lib/librte_ethdev/rte_ethdev.c#L3482
Please add a check to the return value of rte_eth_dev_set_mtu here, and exit the program if the jumbo frame flag is set but is not supported by the NIC.

Reply:
> Is this configuration enough for enabling jumbo packet support? I see some rx port configuration `DEV_RX_OFFLOAD_JUMBO_FRAME` in this Feature list [1]? Also, you can check the usage of this configuration in [2] example.
> 
> [1] https://dpdk.readthedocs.io/en/v17.11/nics/features.html
> [2] https://github.com/DPDK/dpdk/blob/main/examples/ipsec-secgw/ipsec-secgw.c#L2175

I tried the solution in [2] and it seems like the `DEV_RX_OFFLOAD_JUMBO_FRAME` offload flag does the job without the need to use `rte_eth_dev_set_mtu()`  provided the value of rxmode.max_rx_pkt_len is also set in the port configuration. The code in [3] shows that the MTU is implicitly set when the offload is in use.

> The DEV_RX_OFFLOAD_JUMBO_FRAME flag offloads some handling of the jumbo frame to the physical NIC. I'm guessing what it does is forces the hardware to assemble the jumbo frame instead of the PMD (correct me if I'm wrong @twood02 ). This could be used to increase performance. I don't think it's necessary to enable this flag in order to support receiving jumbo frames. We should check if the NIC PMD we are using actually supports the MTU that is passed in. DPDK will tell you if the PMD doesn't have a set_mtu function or if it doesn't support the MTU: https://github.com/DPDK/dpdk/blob/main/lib/librte_ethdev/rte_ethdev.c#L3482
> Please add a check to the return value of rte_eth_dev_set_mtu here, and exit the program if the jumbo frame flag is set but is not supported by the NIC.

Yes, using the offload flag isn't necessary in order to support receiving jumbo frames. However, using it seems like the better way to go. The code in [3] checks whether the device would support the desired jumbo frame length (9618 in this case).

 The support for jumbo frames in pktgen-dpdk is quite similar to what is done in [2]. I've updated the PR to match the method and values used in this patch for pktgen-dpdk [4]. 

[3] https://github.com/DPDK/dpdk/blob/2a2ebeab9e3309ab332baefdbcc1caa7e62ecc82/lib/librte_ethdev/rte_ethdev.c#L1408
[4] http://git.dpdk.org/apps/pktgen-dpdk/commit/app?id=ffc2494bdfcaadb8885b8afcc40be33cf8971372

Reply:
I'm fine with the offload flag - there's a chance that some NIC's won't support this, so we should be wary of that. Btw, how did you test that jumbo frames are properly received and processed? Could you post replication steps?

Reply:
> I'm fine with the offload flag - there's a chance that some NIC's won't support this, so we should be wary of that. Btw, how did you test that jumbo frames are properly received and processed? Could you post replication steps?

Yes, I'm not sure how an unsupported NIC would react to this flag. It looks like some of the sample applications and pktgen-dpdk follow this approach when enabling jumbo frames. 

I tested the recent changes using pktgen-dpdk for traffic generation and the l2fwd example NF for packet processing. I have updated the PR description (Test Plan) to include the commands that I used.

Reply:
Thank you!